modules/ROOT/pages/dw-operators.adoc

```diff
@@ -363,3 +363,359 @@ xref:dataweave-variables.adoc#example_local_variable[Examples: Local DataWeave V
+      "zipCode": "ZZ123"
+    }
+  ]
+}
+----
+
+==== Attribute Selector
+
+This matching selector allows to match any attribute value. +
+
+Example: Update an attribute value and change it to upper case.
+
+.Input
+[source,xml,linenums]
+----
+<user name="leandro"/>
+----
+
+.Source
+[source,weave,linenums]
+----
+%dw 2.0
+output application/json
+---
+payload update {
+    case name at .user.@name -> upper(name)
+}
+----
+
+.Output
+[source,xml,linenums]
+----
+<?xml version='1.0' encoding='UTF-8'?>
+<user name="LEANDRO"/>
+----
+
+==== Dynamic Selector
+
+Any of the given selectors can use an expression to make it dynamic. +
+
+Example: Select and update a value where the field to be updated is in a variable and not hardcoded. In this case from the variable `theFieldName` with value `name`.
+
+.Input
+[source,json,linenums]
+----
+{
+    "name": "Ken",
+    "lastName":"Shokida"
+}
+----
+
+.Source
+[source,weave,linenums]
+----
+%dw 2.0
+output application/json
+var theFieldName = "name"
+---
+payload update {
+    case s at ."$(theFieldName)" -> "Shoki"
+}
+----
+
+.Output
+[source,json,linenums]
+----
+{
+  "name": "Shoki",
+  "lastName": "Shokida"
+}
+----
+
+=== Conditional Update
+
+The update operator has a syntax to write a conditional update if you only want to modify a field under a given condition. +
+
+Example: Update the category based on the name of the user.
+
+.Input
+[source,json,linenums]
+----
+[{"name": "Ken", "age": 30}, {"name": "Tomo", "age": 70}, {"name": "Kajika", "age": 10}]
+----
+
+.Source
+[source,weave,linenums]
+----
+%dw 2.0
+output application/json
+---
+payload map ((user) ->
+    user update {
+        case name at .name if(name == "Ken") -> name ++ " (Leandro)"
+        case name at .name if(name == "Tomo") -> name ++ " (Christian)"
+    }
+)
+----
+
+
+.Output
+[source,json,linenums]
+----
+[
+  {
+    "name": "Ken (Leandro)",
+    "age": 30
+  },
+  {
+    "name": "Tomo (Christian)",
+    "age": 70
+  },
+  {
+    "name": "Kajika",
+    "age": 10
+  }
+]
+----
+
+=== Upserting
+
+The update operator allows to insert a field if it is not present using the `!` symbol at the end of selector expression. When the field is not present the value `null` is going to be bind to the variable. +
+
+Example: Update the category based on the name of the user.
+
+.Input
+[source,json,linenums]
+----
+[{"lastName": "Doe"}, {"lastName": "Parker", "name": "Peter" }]
+----
+
+.Source
+[source,weave,linenums]
+----
+%dw 2.0
+output application/json
+---
+payload  map ((user) ->
+    user update {
+        case name at .name! -> if(name == null) "JHON" else upper(name)
+    }
+)
+----
+
+.Output
+[source,json,linenums]
+----
+[
+  {
+    "lastName": "Doe",
+    "name": "JHON"
+  },
+  {
+    "lastName": "Parker",
+    "name": "PETER"
+  }
+]
+----
+
+
+=== Sugar Syntax
+
+The update operator allows to write a shorter version syntax to avoid some code boilerplate and this is by binding a default variable name `$`. +
```

Comment:
`boilerplate code`? `this is by` -> `by`, `named`

Reply:
Updated.